SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -695,123 +695,96 @@ void CFlowOutput::SetRotatingFrameCoefficients(CConfig *config, CSolver *flow_so
-        if (position!=string::npos) text_line.erase (position,1);
-      }
-      stringstream  point_line(text_line);
-
-      if (geometry->GetnDim() == 2) point_line >> iPoint >> XCoord >> YCoord >> Pressure >> PressureCoeff;
-      if (geometry->GetnDim() == 3) point_line >> iPoint >> XCoord >> YCoord >> ZCoord >> Pressure >> PressureCoeff;
-
-      if (PointInDomain[iPoint]) {
-
-        /*--- Find the vertex for the Point and Marker ---*/
-
-        iMarker = Point2Vertex[iPoint][0];
-        iVertex = Point2Vertex[iPoint][1];
-
-        solver->SetCPressureTarget(iMarker, iVertex, PressureCoeff);
-
+      /*--- remove commas ---*/
+      for (auto& c : text_line) if (c == ',') c = ' ';
+      stringstream point_line(text_line);
+
+      /*--- parse line ---*/
+      unsigned long iPointGlobal;
+      su2double XCoord, YCoord, ZCoord=0, Pressure, PressureCoeff;
+
+      point_line >> iPointGlobal >> XCoord >> YCoord;
+      if (nDim == 3) point_line >> ZCoord;
+      point_line >> Pressure >> PressureCoeff;
+
+      const auto iPoint = geometry->GetGlobal_to_Local_Point(iPointGlobal);
+
+      /*--- If the point is on this rank set the Cp to associated vertices
+       *    (one point may be shared by multiple vertices). ---*/
+      if (iPoint >= 0) {
+        bool set = false;
+        for (auto iMarker = 0u; iMarker < geometry->GetnMarker(); ++iMarker) {
+          const auto iVertex = geometry->nodes->GetVertex(iPoint, iMarker);
+
+          if (iVertex >= 0) {
+            solver->SetCPressureTarget(iMarker, iVertex, PressureCoeff);
+            set = true;
+          }
+        }
+        if (!set)
+          cout << "WARNING: In file " << surfCp_filename << ", point " << iPointGlobal << " is not a vertex." << endl;
       }
-
     }
+  }
 
-    Surface_file.close();
-
-    delete [] Point2Vertex;
-    delete [] PointInDomain;
+  /*--- Compute the pressure difference. ---*/
 
-    /*--- Compute the pressure difference ---*/
+  su2double PressDiff = 0.0;
 
-    PressDiff = 0.0;
-    for (iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {
-      Boundary   = config->GetMarker_All_KindBC(iMarker);
+  for (auto iMarker = 0u; iMarker < geometry->GetnMarker(); ++iMarker) {
 
-      if (config->GetSolid_Wall(iMarker) || (Boundary == NEARFIELD_BOUNDARY)) {
-        for (iVertex = 0; iVertex < geometry->GetnVertex(iMarker); iVertex++) {
+    const auto Boundary = config->GetMarker_All_KindBC(iMarker);
 
-          Normal = geometry->vertex[iMarker][iVertex]->GetNormal();
+    if (config->GetSolid_Wall(iMarker) || (Boundary == NEARFIELD_BOUNDARY)) {
+      for (auto iVertex = 0ul; iVertex < geometry->GetnVertex(iMarker); iVertex++) {
 
-          Cp = solver->GetCPressure(iMarker, iVertex);
-          CpTarget = solver->GetCPressureTarget(iMarker, iVertex);
+        const auto iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
+        if (!geometry->nodes->GetDomain(iPoint)) continue;
```

Comment:
Fai (For anyone's interest): this is the missing check of domain points that is mentioned in the cfd-online post